src/core/ext/filters/client_channel/subchannel.cc

```diff
@@ -970,7 +969,6 @@ void Subchannel::OnConnectingFinished(void* arg, grpc_error* error) {
     c->MaybeStartConnectingLocked();
     GRPC_SUBCHANNEL_WEAK_UNREF(c, "connecting");
   }
-  gpr_mu_unlock(&c->mu_);
   GRPC_SUBCHANNEL_WEAK_UNREF(c, "on_connecting_finished");
```

Comment:
Don't we need to give up the lock before we unref the subchannel?  Otherwise, we might try to unlock after the object is destroyed.

Reply:
Nice catch! This was a typo. I forgot to add the block around it.